src/Handler/TagHandler.php

```diff
@@ -59,10 +98,26 @@ public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cach
      */
     public function invalidateTags(array $tags)
     {
-        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $tags)));
+        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $this->escapeTags($tags))));
         $headers = array($this->tagsHeader => $tagExpression);
         $this->invalidator->invalidate($headers);
 
         return $this;
     }
+
+    /**
+     * Make sure that the tags are valid.
+     *
+     * @param array $tags The tags to escape.
+     *
+     * @return array Sane tags.
+     */
+    protected function escapeTags(array $tags)
```

Comment:
i think protected makes sense here, as an extending class might want to use this for other operations. or a special setup might need additional escaping.